src/upload_all_java_tools.sh

```diff
@@ -32,14 +26,76 @@
 # The script cannot be invoked through a sh_binary using bazel because git
 # cannot be used through a sh_binary.
 
+set -euo pipefail
+
+case "$(uname -s | tr [:upper:] [:lower:])" in
+msys*|mingw*|cygwin*)
+  declare -r is_windows=true
+  ;;
+*)
+  declare -r is_windows=false
+  ;;
+esac
+
+if "$is_windows"; then
+  export MSYS_NO_PATHCONV=1
+  export MSYS2_ARG_CONV_EXCL="*"
+fi
+
 commit_hash=$(git rev-parse HEAD)
 timestamp=$(date +%s)
 bazel_version=$(bazel info release | cut -d' ' -f2)
 
 # Passing the same commit_hash and timestamp to all targets to mark all the artifacts
 # uploaded on GCS with the same identifier.
 for java_version in 9 10; do
-  for target in src:upload_java_tools_java${java_version} src:upload_java_tools_dist_java${java_version}; do
-    bazel run ${target} -- --commit_hash ${commit_hash} --timestamp ${timestamp} --bazel_version ${bazel_version}
-  done
+
+    bazel build //src:java_tools_java${java_version}_zip
+    zip_path=${PWD}/bazel-bin/src/java_tools_java${java_version}.zip
+
+    if "$is_windows"; then
+        # Windows needs "file:///c:/foo/bar".
+        file_url="file:///$(cygpath -m ${zip_path})"
+    else
+        # Non-Windows needs "file:///foo/bar".
+        file_url="file://${zip_path}"
+    fi
+
+    test_name="bazel_java_test_javac${java_version}_${commit_hash}_$(date +%s)"
+    cat >>src/test/shell/bazel/BUILD <<EOF
```

Comment:
Instead of writing a BUILD file from a shell script (which is kinda surprising), how about this:

```
sh_test(
    name="bazel_java_test_with_zip",
    srcs=["bazel_java_test.sh"],
    args=["$(EXTERNAL_JAVA_TEST_FILE"],
    tags=["manual"],  # Only for uploading Java tests                                                    
```

Then you can call it with

```
bazel test --test_output=all :bazel_java_test_with_zip --define=EXTERNAL_JAVA_TEST_FILE=<path. url. whatever.>
```

It's probably a bit more straightforward that way.

Reply:
Now I'm embarrassed I didn't think about this :woman_facepalming:  Thanks!